packages/ovault-evm/contracts/OVaultComposer.sol

```diff
@@ -126,116 +152,197 @@ contract OVaultComposer is IOVaultComposer, ReentrancyGuard {
     }
 
-    /// @dev Permissionless function to retry the message with more gas
-    /// @dev Failure case when there is a LayerZero config issue - ex: dvn config
-    function retry(bytes32 _guid, bytes calldata _extraOptions) external payable nonReentrant {
+    /// @dev If removeExtraOptions is true, only enforcedOptions are used and the sender needs to pay the entire msg.value
+    /// @dev A transaction can never fail due to a lack of extraOptions, unless the OFT is customized in-which case the Composer would need to be customized.
+    function retry(bytes32 _guid, bool removeExtraOptions) external payable nonReentrant {
         FailedMessage memory failedMessage = failedMessages[_guid];
-        if (failedGuidState(_guid) != FailedState.CanOnlyRetry) revert CanNotRetry(_guid);
+        if (_failedGuidState(failedMessage) != FailedState.CanOnlyRetry) revert CanNotRetry(_guid);
 
         SendParam memory sendParam = failedMessage.sendParam;
+        uint256 prePaidMsgValue = failedMessage.msgValue;
 
-        sendParam.extraOptions = _extraOptions;
+        if (removeExtraOptions) {
+            (bool sent, ) = payable(REFUND_OVERPAY_ADDRESS).call{ value: prePaidMsgValue }("");
+            require(sent, "Failed to send Ether");
 
-        delete failedMessages[_guid];
-        _send(failedMessage.oft, sendParam);
+            sendParam.extraOptions = "";
+            prePaidMsgValue = 0;
+        }
+
+        this.sendFailedMessage{ value: msg.value }(_guid, failedMessage.oft, sendParam, prePaidMsgValue, tx.origin);
         emit Retried(_guid, failedMessage.oft);
     }
 
-    /// @dev Retry mechanism for transactions that failed due to slippage. This can revert.
-    /// @dev Failure case when there is a LayerZero config issue - ex: dvn config
-    function retryWithSwap(bytes32 _guid, bytes calldata _extraOptions) external payable {
+    /// @dev Performs the ovault action for a failed swap. If we fail on the send then the GUID enters the retry state for manual execution.
+    function retryWithSwap(bytes32 _guid, bool skipRetry) external payable nonReentrant {
         FailedMessage memory failedMessage = failedMessages[_guid];
-        if (failedGuidState(_guid) != FailedState.CanRetryWithSwap) revert CanNotRetry(_guid);
-
-        SendParam memory sendParam = failedMessage.sendParam;
-        sendParam.extraOptions = _extraOptions;
+        if (_failedGuidState(failedMessage) != FailedState.CanRefundOrRetryWithSwap) revert CanNotSwap(_guid);
+
+        /// @dev Disable refund. If this call reverts refund is still possible.
+        failedMessages[_guid].refundOFT = address(0);
+
+        uint256 srcAmount = failedMessage.refundSendParam.amountLD;
+
+        uint256 vaultAmount = _executeOVaultActionWithSlippageCheck(
+            failedMessage.refundOFT,
+            srcAmount,
+            failedMessage.sendParam.minAmountLD
+        );
+
+        failedMessage.sendParam.amountLD = vaultAmount;
```

Comment:
This isn't persisted in storage (e.g. failedMessages[_guid]), so this is lost when `skipRetry==true` or `this.sendFailedMessage()` reverts.
This function can't be executed again because `failedMessages[_guid].refundOFT = address(0);` so the funds are lost